src/Adapter/PDF/DeliverySlipPdfGenerator.php

```diff
@@ -71,6 +72,10 @@ public function generatePDF(array $orderId)
             throw new RuntimeException($this->translator->trans('The order cannot be found within your database.', [], 'Admin.Orderscustomers.Notification'));
         }
 
+        if (Hook::exec('actionPDFDeliverySlipGenerator', ['order' => $order])) {
+            return;
+        }
+
         $order_invoice_collection = $order->getInvoicesCollection();
 
         $pdf = new PDF($order_invoice_collection, PDF::TEMPLATE_DELIVERY_SLIP, Context::getContext()->smarty);
```

Comment:
```suggestion
        if (Hook::exec('actionPDFDeliverySlipRender', ['order_invoice_list' => $order_invoice_list])) {
            return;
        }

        $pdf = new PDF($order_invoice_collection, PDF::TEMPLATE_DELIVERY_SLIP, Context::getContext()->smarty);
```

I take the same hook as in `src/Adapter/PDF/OrderInvoicePdfGenerator.php` file.

Reply:
Thanks for the feedback @Progi1984 , in fact, at the beginning, I started adding the hook like that, but short after, I realize that the hook for the delievery slip don't really need the invoices array. Invoice are directly related to the invoices pdf but not to the delivery one. This is why I end up passing the actual order object to the hook. What do you think?